tools/cli/go-whisk/whisk/client.go

```diff
@@ -315,11 +315,24 @@ func parseWhiskErrorResponse(resp *http.Response, data []byte, v interface{}) (*
             NO_MSG_DISPLAYED, APPLICATION_ERR)
         return parseSuccessResponse(resp, data, v), whiskErr
     } else {
-        Debug(DbgError, "HTTP response with unexpected body failed due to contents parsing error: '%v'\n", err)
-        errMsg := wski18n.T("The connection failed, or timed out. (HTTP status code {{.code}})",
-            map[string]interface{}{"code": resp.StatusCode})
-        whiskErr := MakeWskError(errors.New(errMsg), resp.StatusCode - 256, DISPLAY_MSG, NO_DISPLAY_USAGE)
-        return resp, whiskErr
+        errorResponse := &RequestErrorResponse{}
+        err := json.Unmarshal(data, errorResponse)
+
+        if err == nil && errorResponse.Response.Result.Error.Error != nil {
```

Comment:
the CLI should not be looking inside of `Error` at all. It should treat `Error` as a JSON object that is opaque.